### PR TITLE
Avoid guid use from deprecated pyarrow.compat

### PR DESCRIPTION
**Pull request by rjzamora**
The `write_to_dataset` parquet code is currently using a deprecated pyarrow API to generate a random file name.  This PR avoids the deprecation issue by using `uuid4.hex` directly.  This PR also follows the lead of [dask#6451](https://github.com/dask/dask/pull/6451) by enabling the user to specify a specific `filename` argument to `write_to_dataset`.  

The dask_cudf to_parquet code will now avoid the uuid4-based approach to file naming.

### CODE REVIEW

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

File: python/cudf/cudf/tests/test_parquet.py [hunk 1]
```diff
@@ -1120,13 +1121,19 @@ def test_parquet_write_partitioned(tmpdir_factory, cols):
     )
     pdf.to_parquet(pdf_dir, index=False, partition_cols=cols)
     gdf = cudf.from_pandas(pdf)
-    gdf.to_parquet(gdf_dir, index=False, partition_cols=cols)
+    gdf.to_parquet(gdf_dir, index=False, partition_cols=cols, partition_file_name=filename)
```
Comment (jakirkham) [inline python/cudf/cudf/tests/test_parquet.py:1124]: Maybe fixes lint

```suggestion
    gdf.to_parquet(
        gdf_dir, index=False, partition_cols=cols, partition_file_name=filename
    )
```

File: python/cudf/cudf/io/parquet.py [hunk 2]
```diff
@@ -320,10 +326,15 @@ def to_parquet(
         if index is None:
             index = True
 
+        # Convert partition_file_name to a call back
+        if partition_file_name:
+            partition_file_name = lambda x: partition_file_name
```
Comment (jakirkham) [inline python/cudf/cudf/io/parquet.py:331]: To silence `flake8` warning about assigning `lambda`s.

```suggestion
            partition_file_name = lambda x: partition_file_name  # noqa: E731
```

File: python/cudf/cudf/tests/test_parquet.py (hunk 1 above)
Comment (rjzamora) [reply python/cudf/cudf/tests/test_parquet.py]: Thanks @jakirkham - I just moved to a new environment and forgot to install my pre-commit hooks :)

**Review by jakirkham — APPROVED** [review]

**Review by galipremsagar — APPROVED** [review]

**Comment by jakirkham** [comment]
rerun tests

**Comment by galipremsagar** [comment]
rerun tests

**Pull request merged**